src/v/kafka/server/handlers/create_topics.cc

```diff
@@ -171,6 +171,36 @@ ss::future<response_ptr> create_topics_handler::handle(
         valid_range_end = unauthorized_it;
     }
 
+    auto additional_resources_func = [&request, has_cluster_auth]() {
```

Comment:
Not saying we should do this but as a nice way to not have these lambdas littered around maybe they can exist as normal functions in another file and the function pointer and arguments can just be passed to audit